packages/shorebird_cli/lib/src/cache.dart

```diff
@@ -162,17 +162,64 @@ abstract class CachedArtifact {
       return false;
     }
 
-    if (checksum == null) {
+    final expectedChecksum = sha256Checksum;
+    if (expectedChecksum == null) {
       logger.detail(
         '''No checksum provided for $fileName, skipping file corruption validation''',
       );
       return true;
     }
 
-    return checksumChecker.checkFile(file, checksum!);
+    return checksumChecker.checkFile(
+      file,
+      checksum: expectedChecksum,
+      algorithm: ChecksumAlgorithm.sha256,
+    );
   }
 
   Future<void> update() async {
+    try {
+      final response = await _makeDownloadRequest();
+      final tempFile = await _downloadFile(response);
+
+      final isZip =
+          response.headers[HttpHeaders.contentTypeHeader] == 'application/zip';
+
+      _verifyChecksum(file: tempFile, responseHeaders: response.headers);
+
+      // Create the directory containing the artifact if it does not already
+      // exist. Failing to do this will cause [renameSync] to throw an
+      // exception.
+      final parentDirectory = file.parent..createSync(recursive: true);
```

Comment:
Will this throw if file.parent already exists?

Reply:
From the `createSync` docs:

> If the directory already exists nothing is done